src/emulation/mod.rs

```diff
@@ -137,18 +139,57 @@ impl SyscallMock {
     }
 }
 
+pub enum Executable {
+    Interpreter(PathBuf),
+    Program(PathBuf),
+}
+
+impl Executable {
+    pub fn to_string_lossy(&self) -> Cow<str> {
+        match self {
+            Executable::Interpreter(path) => path,
+            Executable::Program(path) => path,
+        }
+        .to_string_lossy()
+    }
+}
+
+impl Into<Vec<u8>> for &Executable {
+    fn into(self) -> Vec<u8> {
+        let path = match self {
+            Executable::Interpreter(path) => path,
+            Executable::Program(path) => path,
+        };
+        path.as_os_str().as_bytes().to_vec()
+    }
```

Comment:
We implement `Into<Vec<u8>>` for Executable, because the [CString constructor](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.new) works on anything that implements `Into<Vec<u8>>`. I have some slightly mixed feelings about the fact that this is implemented on a reference, but changing this would mean cloning Executable afaict, which seems worse